reframe/core/schedulers/slurm.py

```diff
@@ -81,7 +81,10 @@ def __init__(self, *args, **kwargs):
     def nodelist(self):
         # Redefine nodelist so as to generate it from the nodespec
         if self._nodelist is None and self._nodespec is not None:
-            self._nodelist = nodelist_expand(self._nodespec)
+            completed = osext.run_command(
+                f'scontrol show hostname {self._nodespec}', log=False
+            )
+            self._nodelist = completed.stdout.splitlines()
```

Comment:
Why don't we fix the `nodelist_expand` and avoid the cost of `scontrol` ?...

Reply:
I think  I will buy that, because Slurm condenses them a bit differently. We can defer the `nodelist_expand` expansion to a later point.